### PR TITLE
Add entry_point for extensions to show display version in cli

### PR DESCRIPTION
**Pull request by davfsa**
For any extension wanting to make use of this, add the following to your `setup.py` or equivalent:

```py
    entry_points={
        "hikari": ["extension_name = module_name"],
    },
```

The version is fetched from the `__version__` variable, accessed through the `module_name`

Note that this is only an suggestion for now and is bound to change until (and if) this pr is merged

### Checklist

- [x] I have run `nox` and all the pipelines have passed.
- [x] I have made unittests according to the code I have added/modified/deleted.

### CODE REVIEW

**Comment by Lunarmagpie** [comment]
Extension dev opinion here: I think this is a very good of doing this. I think you should go forward with this implementation.

File: hikari/cli.py [hunk 1]
```diff
@@ -43,3 +44,13 @@ def main() -> None:
     sys.stderr.write(f"located at {path}\n")
     sys.stderr.write(f"{py_impl} {py_ver} {py_compiler}\n")
     sys.stderr.write(" ".join(frag.strip() for frag in platform.uname() if frag and frag.strip()) + "\n")
+
+    if extensions := importlib.metadata.entry_points().get("hikari"):
+        sys.stderr.write("\nInstalled extensions:\n")
+
+        for extension in extensions:
+            module = extension.load()
+            module_path = os.path.abspath(os.path.dirname(module.__file__))
```
Comment (FasterSpeeding) [inline hikari/cli.py:53]: I would've used pathlib here personally lol but doesn't matter either eay

**Comment by FasterSpeeding** [comment]
Also this would need to be documented somewhere (probably the readme or an extra doc page about extensions)

File: hikari/cli.py [hunk 2]
```diff
@@ -43,3 +44,13 @@ def main() -> None:
...
+            version = getattr(module, "__version__", "unknown version")
```
Comment (FasterSpeeding) [inline hikari/cli.py:54]: have you considered trying `importlib.metadata.version` here?

File: hikari/cli.py [hunk 3]
```diff
@@ -43,3 +44,13 @@ def main() -> None:
     sys.stderr.write(f"located at {path}\n")
     sys.stderr.write(f"{py_impl} {py_ver} {py_compiler}\n")
     sys.stderr.write(" ".join(frag.strip() for frag in platform.uname() if frag and frag.strip()) + "\n")
+
+    if extensions := importlib.metadata.entry_points().get("hikari"):
```
[the remark this answers is not in the document]
Comment (davfsa) [reply hikari/cli.py:48]: Unfortunately `.select` wasn't added until 3.10. Before, 3.10, it returned a normal dict.

Ig I'll have a version check for what call to make

**Comment by davfsa** [comment]
> Also this would need to be documented somewhere (probably the readme or an extra doc page about extensions) 

I initially intended to make this a draft, just to get opinions on it. Will ofc add that :)

**Comment by davfsa** [comment]
> Just wondering if you considered displaying extensions in the printout the Bot classes emit when they're initiated since that seems to be seen by more ppl? 

Would be nice too, will have a look

File: hikari/cli.py (hunk 2 above)
Comment (davfsa) [reply hikari/cli.py:54]: Initially did, but had a dumb mistake and couldnt get it work. Now i realized what that dumb mistake was and switched to it

Thanks for pointing it out and having me have another look at it!

File: hikari/cli.py (hunk 1 above)
Comment (davfsa) [reply hikari/cli.py:53]: I copy pasted this from above, but switched both to make use of pathlib

**Comment by parafoxia** [comment]
Just had a peek at this, and it looks good! I can't see anything wrong with the current implementation.

**Comment by parafoxia** [comment]
> Just wondering if you considered displaying extensions in the printout the Bot classes emit when they're initiated since that seems to be seen by more ppl? 

We'd also need to consider how best to handle this when it comes to custom banners too, as Crescent and Kasai both have their own in place of Hikari's.

If this was a config arg you could pass (which I imagine it probably would be), we'd need to decide how much that key actually handles, such as whether it would format them fully, or provide the option to have a custom delimiter, etc..

**Comment by Lunarmagpie** [comment]
> Just wondering if you considered displaying extensions in the printout the Bot classes emit when they're initiated since that seems to be seen by more ppl?

I think this should be easy to support for custom banners. Just kindly ask all banners to have a new `$-substitution` that gets replaced with the extensions.

**Comment by davfsa** [comment]
Closing for now, maybe revisited at some point in the future

**Pull request closed without merging**